sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedMicroBatchStream.scala

```diff
@@ -38,8 +38,11 @@ private class ChangeFeedMicroBatchStream
   private val containerConfig = CosmosContainerConfig.parseCosmosContainerConfig(config)
   private val partitioningConfig = CosmosPartitioningConfig.parseCosmosPartitioningConfig(config)
   private val changeFeedConfig = CosmosChangeFeedConfig.parseCosmosChangeFeedConfig(config)
-  private val client = CosmosClientCache(clientConfiguration, Some(cosmosClientStateHandle))
-  private val container = ThroughputControlHelper.getContainer(config, containerConfig, client)
+  private val clientCacheItem = CosmosClientCache(
+    clientConfiguration,
+    Some(cosmosClientStateHandle),
+    s"ChangeFeedMicroBatchStream(streamId ${streamId})")
+  private val container = ThroughputControlHelper.getContainer(config, containerConfig, clientCacheItem.client)
```

Comment:
should we also use the Loan pattern here?

Reply:
doesn't work - teh MicroBatchStream is long-running - and has a contract that allows cleaning up resources (in stop())